components/modal/ModalFooterButtons.jsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Box } from '../Box';
+import { ModalFooterButton } from './ModalFooterButton';
+import { deprecateProp } from '../utils';
+
+/** A helper component intended for use inside a `Modal.Footer`. */
+export const ModalFooterButtons = ({ children, commitButton, cancelButton, deleteButton }) => {
+	deprecateProp(
+		commitButton,
+		'The `commitButton` prop has been deprecated in favor of using `Modal.FooterButton` children. See the docs page for our current pattern: https://faithlife.github.io/styled-ui/#/modal',
+	);
+	deprecateProp(
+		cancelButton,
+		'The `cancelButton` prop has been deprecated in favor of using `Modal.FooterButton` children. See the docs page for our current pattern: https://faithlife.github.io/styled-ui/#/modal',
+	);
+	deprecateProp(
+		deleteButton,
+		'The `deleteButton` prop has been deprecated in favor of using `Modal.FooterButton` children. See the docs page for our current pattern: https://faithlife.github.io/styled-ui/#/modal',
+	);
+
+	if ((commitButton || cancelButton || deleteButton) && children) {
+		throw new Error(
+			'`children` cannot be used with `commitButton`, `cancelButton`, or `deleteButton`.',
+		);
+	}
+
+	const buttons = [];
+	if (commitButton) {
+		buttons.push(<LegacyFooterButton buttonObject={commitButton} variant="primary" />);
+		if (cancelButton) {
+			buttons.push(<LegacyFooterButton buttonObject={cancelButton} variant="secondary" />);
+		}
+		if (deleteButton) {
+			buttons.push(<LegacyFooterButton buttonObject={deleteButton} variant="danger" floatAcross />);
+		}
+	} else {
+		if (deleteButton) {
+			buttons.push(<LegacyFooterButton buttonObject={deleteButton} variant="dangerSpecial" />);
+		}
+		if (cancelButton) {
+			buttons.push(<LegacyFooterButton buttonObject={cancelButton} variant="secondary" />);
+		}
+	}
```

Comment:
I think you need `key` props here or every consumer will start seeing an error in the console.

```suggestion
	if (commitButton) {
		buttons.push(<LegacyFooterButton key="commit" buttonObject={commitButton} variant="primary" />);
		if (cancelButton) {
			buttons.push(<LegacyFooterButton key="cancel" buttonObject={cancelButton} variant="secondary" />);
		}
		if (deleteButton) {
			buttons.push(<LegacyFooterButton key="delete" buttonObject={deleteButton} variant="danger" floatAcross />);
		}
	} else {
		if (deleteButton) {
			buttons.push(<LegacyFooterButton key="delete" buttonObject={deleteButton} variant="dangerSpecial" />);
		}
		if (cancelButton) {
			buttons.push(<LegacyFooterButton key="cancel" buttonObject={cancelButton} variant="secondary" />);
		}
	}
```

Reply:
[`4c1dfa3`](https://github.com/Faithlife/styled-ui/pull/529/commits/4c1dfa3c000d3fb99d2e1b474c58df39a6fda0a0)

Reply:
I went with `"commit1"`, `"cancel1"`, `"delete1"`, `"delete2"`, and `"cancel2"`, in case a `commitButton` gets added or removed sometime during the lifecycle, since the `deleteButton` should change when that happens. Doubt that situation would actually occur, but may as well.

Reply:
I don't understand why the different keys are necessary. The code should work just fine with `key="commit"`, `key="cancel"`, and `key="delete"` even if you change the `*button` props throughout the component's lifetime. I think key props are mostly used to [help the DOM reconciliation algorithm](https://reactjs.org/docs/reconciliation.html#recursing-on-children), not avoid rerenders completely.

Reply:
[`485c7e5`](https://github.com/Faithlife/styled-ui/pull/529/commits/485c7e5ae208756b560b58cd8dd7999ebf18975e)